### DISCUSSION TITLE
Guidance for common error messaging?

### DISCUSSION THREAD

**Discussion by mjanssen2021** [twilio-labs/paste#1642] [Q&A] [answered]
We are updating some old error messaging for Authy native app - and want to know if we have any guidelines for error states? 

We are looking at the updated text to be: 
* `Required field missing`
* `Invalid number`
* `Invalid email`

- I checked the[ Paste Product Guideline](https://paste.twilio.design/content/product-style-guide/#common-errors) which didn't have anything on error state guidelines.  
- I checked[ Material's error states](https://material.io/archive/guidelines/patterns/errors.html#errors-user-input-errors) which mostly said to make the error short/concise.

**Comment by serifluous** [comment, marked as answer]
Hey @mjanssen2021 ! Are you looking more for guidance on what the copy should be, or how the error interaction should work?

You could check out this section on Errors for Inputs: https://paste.twilio.design/components/input#errors. There are no _specific_ guidelines for missing or invalid fields, so I'd recommend syncing with @ctecks if there's anything you're particularly concerned about, especially if the guidelines and examples on that page don't give you what you need!

**Comment by ctecks** [comment]
Hi @mjanssen2021 - there's a very brief look at an error notification here https://paste.twilio.design/content/writing-for-components/ that's been unfortunately deprioritized - be sure to ask your manager about hiring some more content resource :). 

As with all content implementations, it really depends on context - where you're flagging the error - is it inline, under a specific field, is it a notification, is it in a modal etc, what's the use case, user need, and positioning in the overall flow have an impact on execution. Can the error be avoided with clearer directions in the UI?

Generally, however, errors should be clear, concise and consistent (at least within the experience, if not as part of the wider product suite.) The user needs to know (again, generally) 
- what the error is [x cannot be done]
- why [optional, but this can help a user troubleshoot - e.g. because the page cannot be reached]
- what to do/how to fix it [this is the most important element - give the user at least one decent option to fix up their issue e.g. refresh the page] - ideal to make this whole 'fix it' phrase actionable, so an inline link or similar

X cannot be done because [location Y] is not responding. _To try again, refresh the page_

Unless the error is catastrophic, avoid saying 'sorry' - it gets in the way of what the user wants, which is a clear indication of what's happened and how to fix it - their goal is their focus, our job is to point them in the right direction and get out of their way.

**Comment by mjanssen2021** [comment]
@ctecks @serifluous 

Thanks for letting me know about these other error content resources! I was thinking that we were bound to have some universal errors across Twilio products, the ones I listed above were for form fields and formatting, so I was double checking to ask if we had any. Seems like we don't for now so this answers my question! But the resources were really helpful and the guidance is a good thing to point to as well. 

Thanks!
